src/vecffe.c

```diff
@@ -1104,7 +1110,19 @@ Obj FuncSMALLEST_FIELD_VECFFE( Obj self, Obj vec)
     q = p;
     for (i = 2; i <= deg; i++)
         q *= p;
-    return INTOBJ_INT(q);
+    return q;
+}
+
+UInt ChooseFieldVecFFE(Obj vec) {
```

Comment:
Please add a proptotype for this somewhere -- either in a header, or locally (in the latter case, might as well make it static, to help the optimizer a bit).